### DISCUSSION TITLE
How to make a background box around a fraction (“bbox”) with Katex?

### DISCUSSION THREAD

**Discussion by q2apro** [KaTeX/KaTeX#2531] [Q&A] [answered]
The `bbox` command is not supported in Katex, see https://katex.org/docs/support_table.html

`bbox` with MathJax renders `\bbox[#e1ffc1,5px]{ \frac{1}{2} }` like:

[![bbox with mathjax][1]][1]

I have tried in Katex the alternative command:

    \colorbox{#e1ffc1}{ \frac{1}{2} }

but it throws the error:

> KaTeX parse error: Can't use function '\frac' in text mode at position 21: …rbox{#e1ffc1}{ \̲f̲r̲a̲c̲{1}{2} }

Likewise `\fcolorbox` does not work.

How to get the background box in Katex around a fraction to work?

<br>

Question was also posted at https://stackoverflow.com/q/63970532/1066234

  [1]: https://i.stack.imgur.com/IlsmD.png

**Comment by ronkok** [comment, marked as answer]
In LaTeX, `\colorbox` puts its contents into text mode. And KaTeX is a bit pickier about text mode than MathJax is. 

You can still put math inside the box, but you have to explicitly shift it from text mode back into math mode with `$…$` delimiters. So, to get your example working, try `\colorbox{#e1ffc1}{$ \frac{1}{2} $}`

**Comment by willymcallister** [comment]
With \colorbox the color specification appears to be hard coded. The allowable arguments are #hexcodes and standard HTML color names. Can you think of a way to make the color responsive to dark/light mode? Find attached some examples. The light mode looks fine, but the dark mode would benefit from a switch to a different background color. The gray background is #808080, the center of the gray scale.

<img width="352" alt="Screen Shot 2020-09-28 at 12 34 14 PM" src="https://user-images.githubusercontent.com/17035787/94477885-539ddf00-0187-11eb-9ad3-c5b619e56c9c.png">

<img width="344" alt="Screen Shot 2020-09-28 at 12 34 31 PM" src="https://user-images.githubusercontent.com/17035787/94477906-5b5d8380-0187-11eb-92a3-8c33be74c2bd.png">

**Reply by edemaine** [reply]
@willymcallister This should probably be a different discussion thread, but I'll answer here anyway: `\htmlClass` is probably what you're looking for. See https://katex.org/docs/supported.html#html
